client/src/Pages/Dallas.js

```diff
@@ -0,0 +1,226 @@
+import { useState, useEffect } from "react";
+
+const Dallas = (props) => {
+  // Initial student in case that you want to update a new student
+  const {
+    initialPost = {
+      id: null,
+      username: "",
+      imageurl: "",
+      alt: "",
+      dish: "",
+      restaurant: "",
+      content: "",
+      city: "",
+      date: "",
+    },
+  } = props;
+
+  // We're using that initial student as our initial state
+  const [post, setPost] = useState(initialPost);
+
+  const [restaurants, setRestaurants] = useState({alias: "", name: "", address: ""});
+
+  //create functions that handle the event of the user typing into the form
+  const handleUsernameChange = (event) => {
+    const username = event.target.value;
+    setPost((post) => ({ ...post, username }));
+  };
+
+  const handleImageURLChange = (event) => {
+    const imageurl = event.target.value;
+    setPost((post) => ({ ...post, imageurl }));
+  };
+
+  const handleAltChange = (event) => {
+    const alt = event.target.value;
+    setPost((post) => ({ ...post, alt }));
+  };
+
+  const handleDishChange = (event) => {
+    const dish = event.target.value;
+    setPost((post) => ({ ...post, dish }));
+  };
+
+  const handleRestaurantChange = (event) => {
+    const restaurant = event.target.value;
+    setPost((post) => ({ ...post, restaurant }));
+  };
+
+  const handleContentChange = (event) => {
+    const content = event.target.value;
+    setPost((post) => ({ ...post, content }));
+  };
+
+  const handleCityChange = (event) => {
+    const city = event.target.value;
+    setPost((post) => ({ ...post, city }));
+  };
+
+  const handleDateChange = (event) => {
+    const date = event.target.value;
+    setPost((post) => ({ ...post, date }));
+  };
+
+  function getRestaurants(e) {
+    e.preventDefault();
+    fetch("/api/location-search", {
+      method: "GET",
+      headers: {"Content-Type": "application/json",},
+    }) .then((response) => response.json())
+    .then((data) => {
+      console.log("data check line 13 ", data.businesses[0]);
+      let fetchRestaurant = {
+        alias: data.businesses[0].alias,
+        name: data.businesses[0].name,
+        address: data.businesses[0].location.display_address
+      }
+      setRestaurants(fetchRestaurant);
+      console.log("checking restaurants line 24 ", restaurants)
+    }) .catch((err) => console.error(err))
+  }
+
+
+  //A function to handle the post request
+  const makePost = (newPost) => {
+    return fetch("/api/blogposts", {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(newPost),
+    })
+      .then((response) => {
+        return response.json();
+      })
+      .then((data) => {
+        console.log("From the post ", data);
+        props.savePost(data);
```

Comment:
I'm not sure where this `savePost` comes from or what it does. What do you want this call to do?

Reply:
I want that to save the updated/edited post. I felt that by naming that "save post", it would make it clear for me what I want to happen at the end of the function. Utilizing props seemed to let me name it anything, but is there a better way to do this?

Reply:
You're right. The name is clear, I'm just not sure where it's "saving" it. But more, I don't see where this function is passed in to the component. I'd expect to see `<Dallas savePost={ (some function to do the saving) }/>` but when I went and looked for this, I don't see the prop. So maybe this is for the future? I just couldn't follow it, so I noted it.

Reply:
ah ok got it. yes, for the future!